src/Extension/FluentVersionedExtension.php

```diff
@@ -145,6 +145,49 @@ public function augmentSQL(SQLSelect $query, DataQuery $dataQuery = null)
         }
     }
 
+    /**
+     * Hook into Versioned::prepareMaxVersionSubSelect - ensure that max version returned by the subquery are limited to
+     * the current locale
+     *
+     * @param SQLSelect $subSelect
+     * @param DataQuery $baseQuery
+     * @param $isCondition
+     */
+    public function augmentMaxVersionSubSelect(SQLSelect $subSelect, DataQuery $baseQuery, $isCondition)
+    {
+        if (!$isCondition) {
+            return;
+        }
+
+        $locale = $this->getDataQueryLocale($baseQuery);
+
+        if (!$locale) {
+            return;
+        }
+
+        $fromTables = $subSelect->getFrom();
+        $baseTable = preg_replace('/' . preg_quote(self::SUFFIX_VERSIONS) . '$/', '', trim(reset($fromTables), '"'));
+        $tableAlias = key($fromTables);
+        $localisedTable = $this->getLocalisedTable($baseTable) . self::SUFFIX_VERSIONS;
+
+        $joinCondition = <<<SQL
+"$tableAlias"."RecordID" = "$localisedTable"."RecordID" AND "$tableAlias"."Version" = "$localisedTable"."Version"
+SQL;
+
+        $subSelect->addLeftJoin(
+            $localisedTable,
+            $joinCondition
+        );
+
+        $locales = "'{$locale->getLocale()}'";
+
+        foreach ($locale->FallbackLocales() as $fallbackLocale) {
+            $locales .= ", '{$fallbackLocale->Locale()->getLocale()}'";
+        }
+
+        $subSelect->addWhere("\"$localisedTable\".\"Locale\" in ({$locales})");
```

Comment:
I hope you don't mind but can we please parameterise this? You can use `DB::placeholders()` to generate the `IN ($placeholders)` component, and then the arguments are the list of locales in fallbacks. That would greatly reduce the risk of SQL injection :)